lib/_http_outgoing.js

```diff
@@ -728,8 +729,28 @@ OutgoingMessage.prototype.setHeaders = function setHeaders(headers) {
     throw new ERR_INVALID_ARG_TYPE('headers', ['Headers', 'Map'], headers);
   }
 
-  for (const key of headers.keys()) {
-    this.setHeader(key, headers.get(key));
+  // Headers object joins multiple cookies with a comma when using
+  // the getter to retrieve the value,
+  // unless iterating over the headers directly.
+  // We also cannot safely split by comma.
+  // To avoid setHeader owerwriting the previous value we push
+  // set-cookie values in array and set them all at once.
+  const headersToSplit = new SafeSet(['set-cookie']);
+  const headersStorage = new SafeMap();
```

Comment:
This will only work if the user calls `setHeaders` only once. Am I right?
If so, can you remove this and get the current value of the header from the already set headers?